### PR TITLE
First steps in tidying image/kernel_read_write tests

### PR DESCRIPTION
**Pull request by ellnor01**
This PR contains a small de-duplicating and movement of functions. No functionality or structure has changed.

Will be pushing several PRs for this issue to keep reviews small and prevent other intermediate changes from getting lost when large blocks of code are moved.

### CODE REVIEW

File: test_conformance/images/kernel_read_write/test_common.cpp [hunk 1]
```diff
@@ -17,3 +33,1514 @@ cl_sampler create_sampler(cl_context context, image_sampler_data *sdata, bool te
...
+void InitFloatCoordsCommon(image_descriptor *imageInfo,
+                           image_sampler_data *imageSampler, float *xOffsets,
+                           float *yOffsets, float *zOffsets, float xfract,
+                           float yfract, float zfract, int normalized_coords,
+                           MTdata d, int lod)
+{
+    size_t i = 0;
+    if (gDisableOffsets)
+    {
+        for (size_t z = 0; z < imageInfo->depth; z++)
```
Comment (jlewis-austin) [inline test_conformance/images/kernel_read_write/test_common.cpp:45]: This requires callers to guarantee depth or height are never zero; is that the intent? I know this has bitten us in other places.

Comment (ellnor01) [reply test_conformance/images/kernel_read_write/test_common.cpp:45]: I've modified this in later patches to account for all image types. In this change the code has only been moved so this is still only run for 3D images.

**Comment by ellnor01** [comment]
Any other comments on this PR? I would like to get it merged so I can add more significant changes on top. 
Also don't want to leave it too long in case it requires a rebase where other changes could get lost.

**Review by jlewis-austin — APPROVED** [review]
LGTM

**Review by bashbaug — APPROVED** [review]
I didn't fully review the content of these changes since it's mostly moving code around, but the moves LGTM.

**Pull request merged**